### PR TITLE
Add rust_g.so

### PR DESCRIPTION
**Pull request by Cyberboss**
Here are the dynamic dependencies for it

```sh
ldd A/rust_g
        linux-gate.so.1 (0xf7f11000)
        libdl.so.2 => /lib/i386-linux-gnu/libdl.so.2 (0xf7e03000)
        librt.so.1 => /lib/i386-linux-gnu/librt.so.1 (0xf7dfa000)
        libpthread.so.0 => /lib/i386-linux-gnu/libpthread.so.0 (0xf7ddb000)
        libgcc_s.so.1 => /lib/i386-linux-gnu/libgcc_s.so.1 (0xf7dbd000)
        libc.so.6 => /lib/i386-linux-gnu/libc.so.6 (0xf7c06000)
        /lib/ld-linux.so.2 (0xf7f13000)
        libm.so.6 => /lib/i386-linux-gnu/libm.so.6 (0xf7bb1000)
```

These are things that should always be present on any sane linux installation (musl libc is reee). So I thought, why not just include it like we do the .dll? Saves on travis times too

This was built out of the tgstation-server docker container (based off of debian:stretch-slim)

### CODE REVIEW

File: code/__DEFINES/rust_g.dm [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // rust_g.dm - DM API for rust_g extension library
-#define RUST_G "rust_g"
+#define RUST_G "rust_g.so"
```
Comment (SpaceManiac) [inline code/__DEFINES/rust_g.dm]: Does this not break Windows?

Comment (Cyberboss) [reply code/__DEFINES/rust_g.dm]: oof

**Comment by Cyberboss** [comment]
This broke BSQL somehow, nani?

**Comment by SpaceManiac** [comment]
doesn't BSQL need libssl

**Comment by Cyberboss** [comment]
Nope, it failed in the build before I removed it

**Comment by optimumtact** [comment]
Don't bundle fucking dynamically linked libs u brainlet, this is going to be a disaster

**Pull request closed without merging**